server/document.js

```diff
@@ -182,24 +182,25 @@ export class NextScript extends Component {
     const pagePathname = getPagePathname(pathname)
 
     return <Fragment>
-      {staticMarkup ? null : <script nonce={this.props.nonce} dangerouslySetInnerHTML={{
-        __html: `
-          __NEXT_DATA__ = ${htmlescape(__NEXT_DATA__)}
-          module={}
-          __NEXT_LOADED_PAGES__ = []
-
-          __NEXT_REGISTER_PAGE = function (route, fn) {
-            __NEXT_LOADED_PAGES__.push({ route: route, fn: fn })
-          }${page === '_error' ? `
-
-          __NEXT_REGISTER_PAGE(${htmlescape(pathname)}, function() {
-            var error = new Error('Page does not exist: ${htmlescape(pathname)}')
-            error.statusCode = 404
-
-            return { error: error }
-          })`: ''}
-        `
-      }} />}
+      {staticMarkup ? null : <script
+        id="server-app-state"
+        type="application/json"
```

Comment:
Can't you use another tag type instead for transferring the app-state? I suppose even though its type is `application/json` some browsers may treat it as an inline script.

Reply:
No. This is an html standard. `<script type=“application/json”>` isn’t JavaScript and CSP defines `script-src` as “Specifies valid sources for JavaScript”. This is also the most valid way for SSR to use CSP.

Reply:
I wouldn't be so optimistic about all browsers implementing this correctly. Anyway, alternatively I could think about using a `data-` attribute on the `head` element to hold the app-state.

Reply:
https://www.drupal.org/node/2513818 

If it means anything to you, a quick google search found that Drupal is using this method.